lib/datasource/gradle/index.js

```diff
@@ -0,0 +1,89 @@
+const got = require('got');
+const { coerce } = require('semver');
+
+module.exports = {
+  getPkgReleases,
+};
+
+async function getPkgReleases(purl, config = {}) {
+  try {
+    const { typeStrategy, digestsStrategy, serviceUrl } = config;
+    let { type, digests } = config;
+
+    if (typeStrategy !== 'auto') {
+      type = typeStrategy;
+    }
+
+    if (digestsStrategy !== 'auto') {
+      digests = digestsStrategy;
+    }
+
+    const response = await got(serviceUrl, {
+      json: true,
+    });
+    const releases = await Promise.all(
+      response.body
+        .filter(release => !release.snapshot && !release.nightly)
+        .filter(
+          release =>
+            // some milestone have wrong metadata and need to be filtered by version name content
+            release.rcFor === '' && !release.version.includes('milestone')
+        )
+        .map(async release => {
+          const entry = {
+            version: coerce(release.version).toString(),
+            downloadUrl: release.downloadUrl,
+            checksumUrl: release.checksumUrl,
+            digests,
+          };
+
+          if (type === 'all') {
+            entry.downloadUrl = replaceType(entry.downloadUrl);
+            entry.checksumUrl = replaceType(entry.checksumUrl);
+          }
+
+          if (digests === 'true') {
+            const checksum = await getChecksum(entry.checksumUrl);
```

Comment:
Won't this mean we're going to download the checksum for every single stable release? Could we instead defer this downloading until the updateDependency function, and therefore only download the checksum for the one version we're updating to?

Reply:
Also, if the checksum was present before, shouldn't it always be present after? Is there any realistic use case where someone would want us to replace the URL but not the checksum?

Reply:
Good point i will change that.